prow/bugzilla/client.go

```diff
@@ -676,13 +678,18 @@ func (c *client) GetComments(bugID int) ([]Comment, error) {
 
 func (c *client) request(req *http.Request, logger *logrus.Entry) ([]byte, error) {
 	if apiKey := c.getAPIKey(); len(apiKey) > 0 {
-		// some BugZilla servers are too old and can't handle the header.
-		// some don't want the query parameter. We can set both and keep
-		// everyone happy without negotiating on versions
-		req.Header.Set("X-BUGZILLA-API-KEY", string(apiKey))
-		values := req.URL.Query()
-		values.Add("api_key", string(apiKey))
-		req.URL.RawQuery = values.Encode()
+		if c.useBearerAuthorization {
```

Comment:
Why not always set both and avoid the need for a flag?

Reply:
I wasn't sure if that will break the call. If it's safe for those headers to co-exists, I am all in to remove the flag.

Reply:
Well, try it out

Reply:
updated.